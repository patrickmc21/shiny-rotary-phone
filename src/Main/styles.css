.App-Main {
  margin: auto;
  height: 680px;
  width: 90vw;
}