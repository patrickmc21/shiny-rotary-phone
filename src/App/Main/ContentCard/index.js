import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentCard = () => {
  return(
    <div>I'm the content card</div>
  )
}

export default ContentCard;