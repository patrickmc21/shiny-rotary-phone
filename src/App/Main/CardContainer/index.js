import React from 'react';
import PropTypes from 'prop-types';
import ContentCard from './ContentCard/index.js';
import './styles.css';

const CardContainer = () => {
  const array = [];
  for (let i =0; i< 100; i++) {
    array.push(i);
  }

  const Cards = array.map(num => <ContentCard />)
  return(
    <section className='Card-Container'>
      {Cards}
    </section>
  )
}

export default CardContainer;