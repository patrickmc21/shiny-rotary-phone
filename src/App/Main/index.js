import React from 'react';
import CardContainer from './CardContainer/index.js';
import PropTypes from 'prop-types';
import './styles.css';

const Main = () => {
  return (
    <main className='App-Main'>
      <CardContainer />
    </main>

  )
}

Main.propTypes = {

};

export default Main;