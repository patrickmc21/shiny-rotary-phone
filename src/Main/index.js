import React from 'react';
import CardContainer from '../CardContainer/index.js';
import PropTypes from 'prop-types';
import './styles.css';

const Main = (props) => {
  const {
    activeCategoryInfo, 
    activeCategoryName,
    updateCurrentCategory, 
    addToFavorites, 
    favorites,
    next,
    previous,
    loading,
  } = props;

  return (
  <div>

  { loading && activeCategoryName !== 'main' ?
    <main className={`App-Main ${activeCategoryName} loading`}>
      <div className={`loading-image ${activeCategoryName}`}>
        <div className='loading-gif'></div>
      </div>
    </main>
    :
    <main className={`App-Main ${activeCategoryName}`}>
    { activeCategoryName === 'main' ?
      <iframe
        title='iframe' 
        width="1300" 
        height="640" 
        src="https://www.youtube.com/embed/Aja1pCUZPso?rel=0&controls=0&showinfo=0&autoplay=1&volume=5" 
        frameBorder="1" 
        allow="autoplay; encrypted-media" 
        allowFullScreen></iframe>
    :
      <CardContainer 
        activeCategoryInfo={activeCategoryInfo}
        activeCategoryName={activeCategoryName}
        addToFavorites={addToFavorites}
        favorites={favorites}
        updateCurrentCategory={updateCurrentCategory}
        next={next}
        previous={previous}/>
    }
    </main>
  }
  </div>

  )
}

Main.propTypes = {
  activeCategoryInfo: PropTypes.array,
  activeCategoryName: PropTypes.string,
  addToFavorites: PropTypes.func,
  favorites: PropTypes.array
};

export default Main;