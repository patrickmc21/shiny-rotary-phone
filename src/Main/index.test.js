import React from 'react';
import { shallow, mount } from 'enzyme';
import Main from './index.js';

describe('Main', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  })
})