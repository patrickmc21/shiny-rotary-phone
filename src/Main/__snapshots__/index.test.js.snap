// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main
        className="App-Main undefined"
>
        <CardContainer
                activeCategoryInfo={undefined}
                activeCategoryName={undefined}
                addToFavorites={undefined}
                favorites={undefined}
                next={undefined}
                previous={undefined}
                updateCurrentCategory={undefined}
        />
</main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CardContainer
          activeCategoryInfo={undefined}
          activeCategoryName={undefined}
          addToFavorites={undefined}
          favorites={undefined}
          next={undefined}
          previous={undefined}
          updateCurrentCategory={undefined}
/>,
        "className": "App-Main undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeCategoryInfo": undefined,
          "activeCategoryName": undefined,
          "addToFavorites": undefined,
          "favorites": undefined,
          "next": undefined,
          "previous": undefined,
          "updateCurrentCategory": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main
          className="App-Main undefined"
>
          <CardContainer
                    activeCategoryInfo={undefined}
                    activeCategoryName={undefined}
                    addToFavorites={undefined}
                    favorites={undefined}
                    next={undefined}
                    previous={undefined}
                    updateCurrentCategory={undefined}
          />
</main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CardContainer
            activeCategoryInfo={undefined}
            activeCategoryName={undefined}
            addToFavorites={undefined}
            favorites={undefined}
            next={undefined}
            previous={undefined}
            updateCurrentCategory={undefined}
/>,
          "className": "App-Main undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeCategoryInfo": undefined,
            "activeCategoryName": undefined,
            "addToFavorites": undefined,
            "favorites": undefined,
            "next": undefined,
            "previous": undefined,
            "updateCurrentCategory": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
